internal/server/server.go

```diff
@@ -266,3 +274,27 @@ func fetchDefaultJobScheduleMapping() map[JobType]string {
 		ExpiringGrantNotification: "0 9 * * *",
 	}
 }
+
+func getAuthInterceptor(config *Config) (grpc.UnaryServerInterceptor, error) {
+	// default fallback to user email on header
+	authInterceptor := withAuthenticatedUserEmail(config.Auth.Default.HeaderKey)
+
+	if config.Auth.Provider == "oidc" {
+		idtokenValidator, err := idtoken.NewValidator(context.Background())
+		if err != nil {
+			return nil, err
+		}
+
+		params := auth.OIDCValidatorParams{
+			Audience:          config.Auth.OIDC.Audience,
+			ValidEmailDomains: config.Auth.OIDC.EligibleEmailDomains,
+			HeaderKey:         config.Auth.Default.HeaderKey,
```

Comment:
can OIDC config has its own `HeaderKey` field so we don't have to add `auth.default.*` when using `auth.provider = oidc`?

Reply:
That make sense. each auth provider should have their own `HeaderKey`.

Reply:
hi @rahmatrhd @ravisuhag, I've refactored header key and (additionally) context key. let me know if you have any concerns